### PR TITLE
BITMAKER-2625 estela: Propose Model and implementation for Activity Menu

### PR DESCRIPTION
**Pull request by raymond1242**
# Description

Currently, there is a disable Activity Menu on estela platform. This menu allows users to watch every process executed on estela. It view records the activity for the whole project

# Issue

[* _Github Issue ID_.](https://tasks.bitmaker.dev/issues/2625)

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] My code follows the style guidelines of this project.
- [x] I have made corresponding changes to the [documentation](https://github.com/bitmakerla/estela/tree/main/docs).
- [x] New and existing tests pass locally with my changes.
- [x] If this change is a core feature, I have added thorough tests.
- [x] If this change affects or depends on the behavior of other _estela_ repositories, I have created pull requests with the relevant changes in the affected repositories. Please, refer to our [official documentation](https://estela.bitmaker.la/docs/).
- [x] I understand that my pull request may be closed if it becomes obvious or I did not perform all of the steps above.

### CODE REVIEW

**Comment by raymond1242** [comment]
@Adriana618 This PR is ready for review

**Comment by raymond1242** [comment]
@emegona This PR is conflicts free

File: estela-api/core/models.py [hunk 1]
```diff
@@ -411,3 +411,28 @@ class UsageRecord(models.Model):
...
+        User,
+        on_delete=models.CASCADE,
+        related_name="activities",
+        help_text="User who performed the activity.",
+    )
+    project = models.ForeignKey(
+        Project,
+        on_delete=models.CASCADE,
+        related_name="activities",
+        help_text="Project to which the activity corresponds.",
```
Comment (mgonnav) [inline estela-api/core/models.py:430]: Let's change this to `Project where the activity was performed.`

File: estela-api/core/models.py [hunk 2]
```diff
@@ -411,3 +411,28 @@ class UsageRecord(models.Model):
...
+    )
+    created = models.DateTimeField(
+        auto_now_add=True, editable=False, help_text="Activity creation date."
```
Comment (mgonnav) [inline estela-api/core/models.py:433]: Let's use `Date when the activity was performed.`. It's more descriptive of what it refers to.

File: estela-api/api/mixins.py [hunk 3]
```diff
@@ -19,3 +20,9 @@ class BaseViewSet(viewsets.GenericViewSet):
     authentication_classes = [TokenAuthentication]
     permission_classes = [IsAuthenticated, IsProjectUser, IsAdminOrReadOnly]
     pagination_class = APIPageNumberPagination
+
+
+class ActivityHandler(viewsets.GenericViewSet):
```
Comment (mgonnav) [inline estela-api/api/mixins.py:25]: Please change this to `class ActivityHandlerMixin:`. Inheriting from `viewsets.GenericViewSet` should not be needed here. Remember to change all the occurrences of the name. Let me know if there are any problems with this change.

**Review by mgonnav — CHANGES_REQUESTED** [review]
I think these changes are extremely similar, not to same pretty much equal, to what was done for the Notifications stuff. In that case, let's hold on on this PR until the notifications changes are merged, make changes here on top of that. I think there's a lot we can refactor betweent he two. Let me know what you think.

**Comment by raymond1242** [comment]
It's correct! Handler and database querys arrñe similar. I'll finish with notifications first

**Comment by raymond1242** [comment]
@mgonnav This PR is ready for review

**Review by mgonnav — APPROVED** [review]

**Pull request merged**